src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/DynamicEvents/DynamicEvents.cs

```diff
@@ -43,69 +43,85 @@ public sealed class DynamicEventSchema
 
         public int MaxOccurrence { get; init; } = 1;
 
-        public static void Set(List<DynamicEventSchema> dynamicEventSchemas, bool allowPartialSchema = true)
+        static DynamicEventSchema()
         {
-            DynamicEventSchemas.Clear();
-            DynamicEventSchema.allowPartialSchema = allowPartialSchema;
-            foreach (DynamicEventSchema dynamicEventSchema in dynamicEventSchemas)
+            // One time add.
+            if (DynamicEventSchemas.Count == 0)
+            {
+                // Add the adaptation schemas.
+                Add(GCDynamicEvents.SizeAdaptationSampleSchema);
+                Add(GCDynamicEvents.SizeAdaptationTuningSchema);
+            }
+        }
+
+        public static void Add(DynamicEventSchema dynamicEventSchema, bool allowPartialSchema = true)
```

Comment:
This method is both compiling the schema and adding it to the list.  I think this might get simpler if you separated those.  If you add a `CompiledSchema Compile(DynamicEventSchema)`, then I think the explicit type initializer could go and `DynamicEventSchemas` could just be initialized to an expression.